programmatic_execution_of_notebooks/DemoQCResultsWithParameters.ipynb

```diff
@@ -0,0 +1,1240 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "DEFAULT_PLOT_HEIGHT <- 7.5\n",
+    "options(repr.plot.width = 16, repr.plot.height = DEFAULT_PLOT_HEIGHT)\n",
+    "theme_set(theme_bw(base_size = 16))\n",
+    "update_geom_defaults('point', list(size = 2))\n",
+    "\n",
+    "#' Returns a data frame with a y position and a label, for use annotating ggplot boxplots.\n",
+    "#'\n",
+    "#' @param d A data frame.\n",
+    "#' @return A data frame with column y as max and column label as length.\n",
+    "#'\n",
+    "get_boxplot_fun_data <- function(d) {\n",
+    "  return(data.frame(y = max(d), label = stringr::str_c('N = ', length(d))))\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Papermill Parameters\n",
+    "\n",
+    "This next cell contains [Papermill parameters](https://papermill.readthedocs.io/en/latest/usage-parameterize.html). You'll notice it has a `parameters` cell tag. You can set cell tags by using the `View` > `Cell Toolbar` > `Tags` menu option.\n",
+    "\n",
+    "Use the `DATASET` parameter to select which dataset you'd like to examine:\n",
+    "\n",
+    "* `simons` for the Simons Genome Diversity Project\n",
+    "* `thousand_genomes` for the 1000 Genomes Phase 3 Release\n",
+    "* `platinum_genomes` for the Deep Variant platinum genomes\n",
+    "\n",
+    "Use the `CSV_OUTPUT_FILE_NAME` parameter to name your CSV output file. If you leave it unset, it will default to `<DATASET>_problem_summary.csv`.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": [
+     "parameters"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "BILLING_PROJECT_ID <- Sys.getenv('GOOGLE_PROJECT')\n",
+    "DATASET <- 'simons'\n",
+    "CSV_OUTPUT_FILE_NAME <- NULL"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Validate Papermill parameters"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "DATASETS = c('simons', 'thousand_genomes', 'platinum_genomes')\n",
+    "\n",
+    "if (!(DATASET %in% DATASETS)) {\n",
+    "    stop(str_glue(DATASET, ' is not a valid dataset name. Valid names are: [', \n",
+    "                  paste(DATASETS, collapse = ', '), ']'))\n",
+    "}\n",
+    "\n",
+    "if (is.null(CSV_OUTPUT_FILE_NAME)) {\n",
+    "    CSV_OUTPUT_FILE_NAME <- str_glue(DATASET, '_problem_summary.csv')\n",
+    "}\n",
+    "\n",
+    "print(DATASET)\n",
+    "print(CSV_OUTPUT_FILE_NAME)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Retrieve sample information"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "retrieve_simons_sample_info <- function() {\n",
+    "    return(bq_table_download(bq_project_query(BILLING_PROJECT_ID, '\n",
+    "    SELECT\n",
```

Comment:
nit: consider separating the constants (the queries) from the code so that there is no need to define these functions.